project/resources/js/Components/DataList.vue

```diff
@@ -8,19 +8,31 @@
                 <slot name="header" :orderBy="orderBy"></slot>
             </thead>
             <tbody>
-                <slot name="body" :items="items"></slot>
+                <slot name="body" :fetchData="fetchData" :items="items"></slot>
             </tbody>
         </table>
 
-        <slot name="footer"></slot>
+        <pagination
+          class="w-10/12"
+          :total-pages="totalPages"
+          :current-page="currentPage"
+          :shouldShowPagination="shouldShowPagination"
```

Comment:
shouldShowPagination, enabledPrevPageButton and enabledNextPageButton can be computed inside pagination component